cmd/agent/main.go

```diff
@@ -69,15 +69,15 @@ func main() {
 			var bindingErr error
 			err = clean.DuplicateBindings(nil)
 			if err != nil {
-				bindingErr = multierror.Append(bindingErr, err)
+				bindingErr = errors.Join(bindingErr, err)
 			}
 			err = clean.OrphanBindings(nil)
 			if err != nil {
-				bindingErr = multierror.Append(bindingErr, err)
+				bindingErr = errors.Join(bindingErr, err)
 			}
 			err = clean.OrphanCatalogBindings(nil)
 			if err != nil {
-				bindingErr = multierror.Append(bindingErr, err)
+				bindingErr = errors.Join(bindingErr, err)
 			}
 			err = bindingErr
```

Comment:
Since from the doc

> Join returns an error that wraps the given errors. Any nil error values are discarded. Join returns nil if every value in errs is nil. 

I guess this be even simplified/shortened with:

```go
bindingErr := errors.Join(
    clean.DuplicateBindings(nil),
    clean.OrphanBindings(nil),
    clean.OrphanCatalogBindings(nil),
)
```

Reply:
I think this looks really clean. I've changed it in a few places to use this format.

Reply:
There is however a __significant__ semantic difference in the execution.

The old code __stops on the first error__.

The simplified code on the other hand always __runs all functions__, even when earlier functions returned errors.

Is that something we want or can deal with ?

Reply:
Oops. I thought the original code stopped on first error. Not so, collecting all of them is the point.
So yes, this way looks definitely cleaner.